src/geometry/euclidean/basic.lean

```diff
@@ -197,7 +197,8 @@ begin
     { rw norm_eq_zero at hy,
       rw [hy, inner_zero_right, zero_mul, neg_zero] } },
   { field_simp [h],
-    ring }
+    rw [mul_assoc ∥x∥ ∥y∥ _, ← mul_assoc ∥y∥ _ _, mul_comm ∥y∥ ∥x∥, mul_assoc ∥x∥ _ _,
```

Comment:
Strangely `ring` stopped working here. Even more strangely, `ring; ring` works, but I did the proof by hand.

Reply:
An `abba` tactic?

@digama0, should we bother to diagnose this example of non-idempotence in `ring`?